cpp/ql/src/semmle/code/cpp/dataflow/internal/FlowVar.qll

```diff
@@ -243,6 +244,54 @@ private module PartialDefinitions {
     }
   }
 
+  class SmartPointerPartialDefinition extends PartialDefinition {
+    Variable pointer;
```

Comment:
I've lost track of how everything works in `PartialDefinition`, but I suspect it may be hurting more than helping to require a `Variable` here. It's no longer required in `VariablePartialDefinition`, despite the name, and it only seems to be used for the deprecated members.

Reply:
It's used by `partiallyDefinesVariableAt`. This predicate is used by `BlockVar::definedPartiallyAt`, and it's behavior is important for the `Node -> FlowVar -> VariableAccess` case in `simpleLocalFlowStep`. Specifically, it's used to get flow out of a partial definition and into variable accesses and parameters. It might be possible to refactor this to something simpler, but I think it's a non-trivial change.